### DISCUSSION TITLE
efficient settings to solve a sparse problem using ginkgo

### DISCUSSION THREAD

**Discussion by jordanpui** [ginkgo-project/ginkgo#1282] [Q&A]
Hi All,

I am having a problem of Ax=b and I use CG to solve it.
A is a very sparse matrix, which have millions of row but each row only have very few items (tens). The matrix is also symmetric, where diagonal is full. b is also full. Currently, I use jacobi preconditioner to speedup, CsrMatrix to represent A,1D Dense Matrix to represent x and b. It takes quite some time to solve with epilson as 0.001. I wonder if there are suitable APIs to improve the efficency.

**Comment by upsj** [comment]
Cg and Jacobi are both components that only operate on small neighborhoods in the underlying mesh (if you come from the FEM perspective), so it is quite possible that they do not work well in your case. You could consider trying an incomplete Cholesky preconditioner instead, that operates non-locally.
